pkgs/objective_c/lib/src/ns_enumerator.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2025, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'internal.dart';
+import 'objective_c_bindings_generated.dart';
+
+class _NSEnumeratorAdapter implements Iterator<ObjCObject> {
+  final NSEnumerator enumerator;
+  ObjCObject? _current;
+
+  _NSEnumeratorAdapter(this.enumerator);
+
+  @override
+  ObjCObject get current => _current!;
+
+  @override
+  @pragma('vm:prefer-inline')
+  bool moveNext() {
+    _current = enumerator.nextObject();
+    return _current != null;
+  }
+}
+
+extension NSEnumeratorToAdapter on NSEnumerator {
+  /// Wraps this [NSEnumerator] in an adapter that implements [Iterator].
+  Iterator<ObjCObject> toDart() => _NSEnumeratorAdapter(this);
```

Comment:
I have a general question here, by exposing the `toDart` version as an `Iterator` (or any other normal Dart class), we're limiting the ability of the user to release the underlying object after use manually, if it's the exact same underlying object, then I would like to have `asDart` instead of `toDart`.

So strings would get a `toDart` and I assume mutable lists would get a `asDart`.